These functions are declared external in a header file, then defined with
static, the mismatch is an error with gcc4

--- GConf-1.0.9/backends/bdb.c.agriffis	2006-02-21 08:55:42.000000000 -0500
+++ GConf-1.0.9/backends/bdb.c	2006-02-21 08:55:51.000000000 -0500
@@ -331,7 +331,7 @@
 
 /* } */
 
-static guint32
+guint32
 get_dir_id (BDB_Store * bdb, const char *dir)
 {
   int ret;
@@ -405,7 +405,7 @@
 
 /* { Functions to create directories and maintain the dir hierarchy */
 
-static guint32
+guint32
 get_or_create_dir (BDB_Store * bdb, const char *dir)
 {
   guint32 parent_id;
